### ISSUE TITLE
Request of implantmentation of lgamma function

### ISSUE THREAD

**Issue by c0nleyinnnn** [ml-explore/mlx#2050]
I am trying to port an mlx implementation of the SCVI module of scvi-tools (a comprehensive de-batching approach using VAE models), which currently overcomes the problem of mlx.array not supporting scipy sparse matrices by converting the sparse matrices to dense matrices, but in practice I can only use python's internal lanczos approximation due to the lack of a native C++ backend lgamma function (something like torch.lgamma or jax.scipy.special.gamma) which leads to a severely limited code with high training loss, and I would like to request the implementation of a built in gamma as well as the lgamma function since I am completely lacking in C code capabilities.

**Comment by c0nleyinnnn** [comment]
[https://github.com/scverse/scvi-tools/issues/3268](url)

**Comment by AvikantSrivastava** [comment]
@awni I am interested, can I pick this one?

**Comment by c0nleyinnnn** [comment]
> [@awni](https://github.com/awni) I am interested, can I pick this one?


plz plz plz plz plz :)

**Comment by ColtAllen** [comment]
> [@awni](https://github.com/awni) I am interested, can I pick this one?

Any movement on this? Otherwise I'll try my hand at vibe-coding something out.

The PyMC library for Bayesian modeling supports MLX as a backend now, and is [attracting user interest](https://discourse.pymc.io/t/how-to-use-pymc-with-mlx/17394/6). MLX speedups scale with dataset size, which is often enormous for the CLV models in [PyMC-Marketing](https://github.com/pymc-labs/pymc-marketing). All CLV models contain log-Gamma in their loss functions, so if we get this added, Bayesian modeling with MLX could really start taking off 🚀

**Comment by c0nleyinnnn** [comment]
> > [@awni](https://github.com/awni) I am interested, can I pick this one?
> 
> Any movement on this? Otherwise I'll try my hand at vibe-coding something out.
> 
> The PyMC library for Bayesian modeling supports MLX as a backend now, and is [attracting user interest](https://discourse.pymc.io/t/how-to-use-pymc-with-mlx/17394/6). MLX speedups scale with dataset size, which is often enormous for the CLV models in [PyMC-Marketing](https://github.com/pymc-labs/pymc-marketing). All CLV models contain log-Gamma in their loss functions, so if we get this added, Bayesian modeling with MLX could really start taking off 🚀

please please please
